### ISSUE TITLE
Not working with Julia 1.0

### ISSUE THREAD

**Issue by kukkuk** [docopt/DocOpt.jl#25]
This library doesn't seem to work with Julia 1.0:

```
julia> using DocOpt
[ Info: Precompiling DocOpt [968ba79b-81e4-546f-ab3a-2eecfa62a9db]
ERROR: LoadError: UndefVarError: iteratorsize not defined
Stacktrace:
 [1] getproperty(::Module, ::Symbol) at ./sysimg.jl:13
 [2] top-level scope at none:0
 [3] include at ./boot.jl:317 [inlined]
 [4] include_relative(::Module, ::String) at ./loading.jl:1038
 [5] include(::Module, ::String) at ./sysimg.jl:29
 [6] top-level scope at none:2
 [7] eval at ./boot.jl:319 [inlined]
 [8] eval(::Expr) at ./client.jl:389
 [9] top-level scope at ./none:3
in expression starting at /home/barry/.julia/packages/DocOpt/QelWY/src/DocOpt.jl:117
ERROR: Failed to precompile DocOpt [968ba79b-81e4-546f-ab3a-2eecfa62a9db] to /home/barry/.julia/compiled/v1.0/DocOpt/ZSrJU.ji.
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] macro expansion at ./logging.jl:313 [inlined]
 [3] compilecache(::Base.PkgId, ::String) at ./loading.jl:1184
 [4] _require(::Base.PkgId) at ./logging.jl:311
 [5] require(::Base.PkgId) at ./loading.jl:852
 [6] macro expansion at ./logging.jl:311 [inlined]
 [7] require(::Module, ::Symbol) at ./loading.jl:834
```

I'm very new to Julia, so I might be doing something wrong. If so, please point me in the right direction.

**Comment by arbenede** [comment]
Same problem with Julia 1.0 on Windows 10

**Comment by mlhetland** [comment]
You're not doing anything wrong – the code is not compatible with 1.0. In this case, the use of `iteratorsize` should be replaced with `IteratorSize`. There are also uses of the old iterator protocol, for example, and `Base.Test` rather than `Test`, and various other no longer functioning uses of functions like `reduce`, etc. I started fixing it (including `if VERSION ≥ v"0.7"` to keep compatibility), but it turned out to be a bit more than I anticipated, so I dropped it for now. If the devs don't have the capacity to fix it, I might have another look later.

**Comment by yadongli** [comment]
I just submitted a pull request which should fix make it 0.7/1.0 compatible.

**Comment by scls19fr** [comment]
Hi @yadongli where is your PR?

**Comment by waldyrious** [comment]
> Hi @yadongli where is your PR?

That would be #26. It looks like this issue can be closed.

**Comment by scls19fr** [comment]
@staticfloat could you close if that's fixed by #26 ?
